fix icon shrink and use avatar for logo and icons

ehconitin opened the pull request:
- Fixed icon shrinking on tabs. The introduction of EllipsisDisplay takes 100% of the width, thus shrinking the icons.
- Removed scroll wrapper tablist. This was removed in #9016 but reintroduced in #9089. This reintroduction made the dark border below the active tab disappear. 
- Used Avatar for icon and logo rendering following the changes made in #9093

ijreilly commented on packages/twenty-front/src/modules/ui/layout/tab/components/Tab.tsx:
```diff
@@ -92,8 +96,24 @@ export const Tab = ({
+            <Avatar
+              Icon={Icon}
+              size="md"
+              placeholder=""
```
from Avatar definition

```
const placeholderChar = placeholder?.[0]?.toLocaleUpperCase();
  const showPlaceholder =
    noAvatarUrl || invalidAvatarUrls.includes(avatarImageURI);
(...)
{Icon ? (
        <Icon
          color={iconColor ? iconColor : 'currentColor'}
          size={theme.icon.size.xl}
        />
      ) : showPlaceholder ? (
        placeholderChar
      ) : (
        <StyledImage src={avatarImageURI} onError={handleImageError} alt="" />
      )}
```

I fear we end up dispaying 'undefined' ?

ehconitin replied:
fixed! realized it after grep's review :)

charlesBochet commented:
@ehconitin why do we remove the scrollwrapper in tab list? we want it in case we have too many tabs and need horizontal scrolling isn't it?

ehconitin commented:
I reverted the tabList changes from #8726 and wrapped the tabList in WorkflowEditActionFormServerlessFunction in a container(as done elsewhere in the codebase).
Added scrollWrapper back similar to #9089!

charlesBochet commented:
That's a mistake, we still need the scrollWrapper!

ehconitin commented:
@charlesBochet added it back! ready for review!

charlesBochet commented:
Discussed this one with @Bonapara to understand the change:
- The current Avatar component that is able to take a logo, an icon, should be renamed to Icon
- we should introduce an other sub-component called IconAvatar that should only take logo or placeholder into props
- current Icon should become IconTablerIcon
- we will introduce other types of icons later such as: IconColor, IconFilledIcon, IconEmoji
